.github/workflows/ci.yml

```diff
@@ -0,0 +1,106 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+
+  merge_group:
+
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  prefetch:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.1.1
+
+      - name: Update pip
+        run: |
+          pip install --upgrade pip
+          pip --version
+
+      - name: Install poetry
+        run: |
+          pip install --upgrade poetry
+          poetry --version
+
+      - name: Setup Python
+        uses: actions/setup-python@v5.0.0
+        with:
+          python-version: 3.11.7
+          cache: poetry
+          cache-dependency-path: '**/poetry.lock'
```

Comment:
does this mean any poetry.lock in the current repo?

Reply:
Yup.